lib/puppet/file_serving/content.rb

```diff
@@ -23,7 +23,7 @@ def self.from_binary(content)
 
   # This is no longer used, but is still called by the file server implementations when interacting
   # with their model abstraction.
-  def collect(source_permissions = nil)
+  def collect
```

Comment:
Are you proposing to revert the changes to `metadata.rb` and most of the changes to `terminus_helper.rb`? I'm not sure `Puppet::FileServing::Metadata#collect` and `Puppet::FileServing::Content#collect` are a public interface we need to preserve; they're not even part of `Puppet::FileServing::Base`. Putting this in `4.1.0` seems reasonable to me.